src/status_im/protocol/web3/delivery.cljs

```diff
@@ -53,16 +53,11 @@
 (defonce pending-message-callback (atom nil))
 (defonce recipient->pending-message (atom {}))
 
-(defn set-pending-mesage-callback!
-  [callback]
-  (reset! pending-message-callback callback))
+(def ^:private pending-message-queue (async/chan 100))
 
-(defn add-pending-message!
-  [web3 {:keys [type message-id requires-ack? to ack?] :as message}]
-  {:pre [(valid? :protocol/message message)]}
-  (go
-    (debug :add-pending-message! message)
-    ;; encryption can take some time, better to run asynchronously
+(async/go-loop [[web3 {:keys [type message-id requires-ack? to ack?] :as message}]
```

Comment:
Having a top-level go-loop isn't a problem ?

Reply:
Nope top level `go-loop` is just a shortcut for `(go (loop ...))` - important thing is that only one go block is established, opposed to always creating new `go` blocks inside the loop periodically (as was the case before), all of those `go` blocks creating channels which are never captured and closed.
Although those inner blocks/channels should be theoretically garbage collected over time, it's more tricky with the code expanding macro like `go` and overall nesting go blocks is a bad practice which has no particular benefit.

Reply:
Agree with this, I guess one enhancement would be to make sure it is only initialized once, a like component mount. But this works and there is no reason to have it toggle-able either as far as I can tell.

Reply:
@janherich how does figwheel work with it? i expect figwheel to run this on each reload.

Reply:
@dmitryn TBH I don't know, I always re-start the app when making changes because there are many other things which require state re-initalisation (like loading commands), so I don't trust figwheel reloading except when making simple UI styling changes (it's great for that).
Maybe `defonce` will be better in regard to figwheel.